lib/beaker/host/unix/exec.rb

```diff
@@ -49,20 +49,20 @@ def reboot(wait_time=10, max_connection_tries=9, uptime_retries=18)
       sleep(reboot_sleep)
 
       exec(Beaker::Command.new('/bin/systemctl reboot -i || reboot || /sbin/shutdown -r now'), :expect_connection_failure => true)
-    rescue Beaker::Host::RebootFailure => e
+    rescue RuntimeError => e
+      raise Beaker::Host::RebootFailure, "Unexpected exception in reboot: #{e.message}"
+    rescue ArgumentError => e
+      raise Beaker::Host::RebootFailure, "Unable to parse time: #{e.message}"
+    rescue => e
+      # Yes, this isn't best practice but rebooting is weird and we really want
+      # to make sure we retry if something went wrong (no matter what it was).
```

Comment:
Since this patch removes the explicit rescue for `Beaker::Host::RebootFailure` and this rescues anything that's left, the `Beaker::Host::RebootFailure` raised on line 34 will end up here.  Is that intentional?

(In case the `rescue RuntimeError` was meant to catch that: `Beaker::Host::RebootFailure` inherits from Ruby's `StandardError`.)

Reply:
@op-ct Yeah, the error was getting caught earlier and missing the `retry`. When I ran through and hit some error cases, the resulting message was clear enough (gets passed through the `raise`) that it was easy to figure out what went wrong.

Reply:
@trevor-vaughan: Can you provide an example (or a way to make it fail in that manner)?  

I don't see how the earlier rescues here could manage to catch a `Beaker::Host::RebootFailure`; it doesn't inherit from either `RuntimeError` or `ArgumentError` (the lineage is [`RebootFailure < CommandFailure < StandardError`](https://github.com/voxpupuli/beaker/blob/90a837cff51b4d9a6487ebb3c961596201afd632/lib/beaker/host.rb#L20-L21)). That suggests that whatever error cases you encountered either didn't involve a `Beaker::Host::RebootFailure` or didn't occur in this block.

Reply:
@op-ct Run the `simp-core` tests about 50 times over a week, watch them fail in random ways and have this be the only scenario that actually works (true story).